action/protocol/staking/handlers.go

```diff
@@ -55,13 +52,17 @@ func (p *Protocol) handleRestake(ctx context.Context, act action.Action, sm prot
 func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.CandidateRegister, sm protocol.StateManager) (*action.Receipt, error) {
 	actCtx := protocol.MustGetActionCtx(ctx)
 	blkCtx := protocol.MustGetBlockCtx(ctx)
-	_, err := getCandidate(sm, actCtx.Caller)
-	if err == nil || errors.Cause(err) != state.ErrStateNotExist {
-		return nil, errors.New("already exist")
+
+	if p.inMemCandidates.ContainsName(act.Name()) {
+		return nil, ErrAlreadyExist
+	}
+
+	if p.inMemCandidates.ContainsOwner(act.OwnerAddress()) {
+		return nil, ErrAlreadyExist
 	}
 
-	if p.inMemCandidates.Contains(act.Name()) {
-		return nil, errors.New("name already in use")
+	if p.inMemCandidates.ContainsOperator(act.OperatorAddress()) {
+		return nil, ErrAlreadyExist
 	}
```

Comment:
duplicate checking with https://github.com/iotexproject/iotex-core/pull/1955/files#diff-6ee6adb0e34757ff08d9b44ca73da49eR67 ?

Reply:
updated in latest push